homeassistant/components/sensor/metoffice.py

```diff
@@ -0,0 +1,159 @@
+"""
+Support for UK Met Office weather service.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.metoffice/
+"""
+
+import logging
+from datetime import timedelta
+
+import voluptuous as vol
+
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import (
+    CONF_MONITORED_CONDITIONS, TEMP_CELSIUS, STATE_UNKNOWN, CONF_NAME,
+    ATTR_ATTRIBUTION)
+from homeassistant.helpers.entity import Entity
+from homeassistant.util import Throttle
+import homeassistant.helpers.config_validation as cv
+
+_LOGGER = logging.getLogger(__name__)
+
+REQUIREMENTS = ['datapoint==0.4.3']
+
+CONF_ATTRIBUTION = "Data provided by the Met Office"
+CONF_MO_API_KEY = 'api_key'
+
+SCAN_INTERVAL = timedelta(minutes=35)
+LAST_UPDATE = 0
+
+# Sensor types are defined like: Name, units
+SENSOR_TYPES = {
+    'name': ['Station Name', None],
+    'weather': ['Weather', None],
+    'temperature': ['Temperature', TEMP_CELSIUS],
+    'feels_like_temperature': ['Feels Like Temperature', TEMP_CELSIUS],
+    'wind_speed': ['Wind Speed', 'mps'],
+    'wind_direction': ['Wind Direction', None],
+    'wind_gust': ['Wind Gust', 'mps'],
+    'visibility': ['Visibility', 'km'],
+    'uv': ['UV', None],
+    'precipitation': ['Probability of Precipitation', '%'],
+    'humidity': ['Humidity', '%']
+}
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Optional(CONF_NAME, default=None): cv.string,
+    vol.Required(CONF_MO_API_KEY): cv.string,
+    vol.Required(CONF_MONITORED_CONDITIONS, default=[]):
+        vol.All(cv.ensure_list, [vol.In(SENSOR_TYPES)]),
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup the sensor platform."""
+    import datapoint as dp
+    datapoint = dp.connection(api_key=config.get(CONF_MO_API_KEY))
+
+    if None in (hass.config.latitude, hass.config.longitude):
+        _LOGGER.error("Latitude or longitude not set in Home Assistant config")
+        return False
+
+    try:
+        site = datapoint.get_nearest_site(longitude=hass.config.longitude,
+                                          latitude=hass.config.latitude)
+    except dp.exceptions.APIException as err:
+        _LOGGER.error("Received error from Met Office Datapoint: %s", err)
+        return False
+
+    if not site:
+        _LOGGER.error("Unable to get nearest Met Office forecast site")
+        return False
+    else:
```

Comment:
We can drop the `else` and adjust the indent since the if statement returns `False`.